htdocs/web_portal/controllers/service_group/add_service_group.php

```diff
@@ -67,16 +69,30 @@ function submit($user) {
     }
 }
 
-function draw($user) {
+function draw($user)
+{
     try {
-        // If the user is registered they're allowed to add a service group
+        $hasRolesAssigned = false;
+        $isUserValid = false;
+        $hasAdminCredentials = $user->isAdmin();
+
         if (is_null($user)) {
             throw new \Exception("Unregistered users can't create service groups.");
         }
 
+        $roleService = \Factory::getRoleService();
+        $hasRolesAssigned = $roleService->getUserRoles($user);
+
+        $isUserValid = $hasAdminCredentials ? true : !empty($hasRolesAssigned);
```

Comment:
what is actually happening on this line?

Reply:
Case1: User is NOT allowed to add new SG's. If a User have no roles assigned.
Case2: User is allowed to add new SG's. If a user is registered and have at least one role assigned with 'STATUS_GRANTED'.

(OR)
 
If a user is an Admin -> they are allowed to ADD new service group's.

That code can help to decide `isUserValid` to perform the task we need. In our case that would be case 1 and case 2.

Reply:
i was more wondering how the " ? true : " works - i'm not doubting it doesn't work, just wondering how

Reply:
Cool. So, I am using ternary operator here. The way it works is "variableName = conditionToEvaluate ? trueBlock : falseBlock". It is a shorthand notation for if...else  like
`
if (conditionToEvaluate) {
    trueBlock
} else {
    falseBlock
}
`
Does that make sense now ?

Reply:
ah right, thanks